go/spec/spec.go

```diff
@@ -91,21 +82,27 @@ func ForDataset(spec string) (Spec, error) {
 
 // ForDatasetOpts parses a spec for a Dataset.
 func ForDatasetOpts(spec string, opts SpecOptions) (Spec, error) {
-	dbSpec, dsName, err := splitDatabaseSpec(spec)
+	dbSpec, pathStr, err := splitDatabaseSpec(spec)
+
+	sp, err := newSpec(dbSpec, opts)
 	if err != nil {
 		return Spec{}, err
 	}
 
-	if !datasetRe.MatchString(dsName) {
-		return Spec{}, fmt.Errorf("Dataset %s must match %s", dsName, datasetRe.String())
-	}
-
-	sp, err := newSpec(spec, dbSpec, opts)
+	path, err := NewAbsolutePath(pathStr)
 	if err != nil {
 		return Spec{}, err
 	}
 
-	sp.DatasetName = dsName
+	if path.Dataset == "" {
+		return Spec{}, errors.New("dataset name required for dataset spec")
+	}
+
+	if !sp.Path.Path.IsEmpty() {
```

Comment:
Seems like this should be `!path.Path.IsEmpty()`. `sp.Path` hasn't been set yet. Is there no test that catches this?